### PR TITLE
core-react `Dialog` deprecation

### PR DESCRIPTION
**Pull request by AntanasBuk**
## Changes

Deprecated core-react `Dialog` and `GlobalDialog` as well as appui-react `ModelessDialog` (weird that it was there and not in core-react). Suppressed relevant  deprecation linting errors.

## Testing

N/A

### CODE REVIEW

File: ui/core-react/src/core-react/dialog/Dialog.tsx [hunk 1]
```diff
@@ -132,12 +134,15 @@ interface DialogState {
 /**
  * Dialog React component with optional resizing and dragging functionality
  * @public
+ * @deprecated in 4.2. Use [Dialog]($itwinui-react) instead.
```
[the remark this answers is not in the document]
Comment (AntanasBuk) [reply ui/core-react/src/core-react/dialog/Dialog.tsx]: Done

**Comment by raplemie** [comment]
Given the issues with iTwinUI dialog for resize and dragging, let's hold on this for now.

**Comment by raplemie** [comment]
Review issue like #662 before closing/continuing.

**Comment by GerardasB** [comment]
Closing until we decide to continue with Dialog deprecation. Seems like iTwinUI 3.x is a prereq for this: https://github.com/iTwin/appui/pull/666

**Pull request closed without merging**